File: scripts/dataframe_to_coverage.py
from helpers import read_file, file_to_grange, init_ray; init_ray(snakemake.wildcards)

from pyrle import PyRles

from time import time
import datetime
import numpy as np

chip = file_to_grange(snakemake.input[0], dtype=np.int64)

print(chip)
start = time()
grles = PyRles(chip, stranded=True)
end = time()
total = end - start

total_dt = datetime.datetime.fromtimestamp(total)

minutes_seconds = total_dt.strftime('%-M.%-S.%f')

open(snakemake.output["timing"], "w+").write(minutes_seconds)

print(grles)

File: scripts/dataframe_to_coverage.R
source("scripts/helpers.R")
library(GenomicRanges)
library(data.table)

## chip = import()
f = snakemake@input[[1]]

gr = file_to_grange(f)

print("Starting to create Rles")

start.time <- Sys.time()
chip_list = coverage(gr)

end.time <- Sys.time()


time.taken <- end.time - start.time

time.taken <- as.numeric(time.taken, units="secs")

write(time.taken, snakemake@output[["timing"]])
capture.output(print(chip_list), file=snakemake@output[["preview"]])
